src/dodal/devices/oav/oav_to_redis_forwarder.py

```diff
@@ -66,68 +75,77 @@ def __init__(
             redis_db: int           which redis database to connect to, defaults to 0
             name: str               the name of this device
         """
+        self.counter = epics_signal_r(int, f"{prefix}CAM:ArrayCounter_RBV")
+
         self._sources = DeviceVector(
             {
-                Source.FULL_SCREEN.value: epics_signal_r(
-                    str, f"{prefix}XTAL:MJPG_URL_RBV"
-                ),
-                Source.ROI.value: epics_signal_r(str, f"{prefix}MJPG:MJPG_URL_RBV"),
+                Source.ROI.value: OAVSource(f"{prefix}MJPG:", "roi"),
+                Source.FULL_SCREEN.value: OAVSource(f"{prefix}XTAL:", "fullscreen"),
             }
         )
         self.selected_source = soft_signal_rw(Source)
 
-        with self.add_children_as_readables():
-            self.uuid, self.uuid_setter = soft_signal_r_and_setter(str)
-
         self.forwarding_task = None
         self.redis_client = StrictRedis(
             host=redis_host, password=redis_password, db=redis_db
         )
 
-        self._stop_flag = False
+        self._stop_flag = asyncio.Event()
 
         self.sample_id = soft_signal_rw(int, initial_value=0)
 
-        # The uuid that images are being saved under, this should be monitored for
-        # callbacks to correlate the data
-        self.uuid, self.uuid_setter = soft_signal_r_and_setter(str)
+        with self.add_children_as_readables():
+            # The uuid that images are being saved under, this should be monitored for
+            # callbacks to correlate the data
+            self.uuid, self.uuid_setter = soft_signal_r_and_setter(str)
 
         super().__init__(name=name)
 
-    async def _get_frame_and_put_to_redis(self, response: ClientResponse):
-        """Converts the data that comes in as a jpeg byte stream into a numpy array of
-        RGB values, pickles this array then writes it to redis.
-        """
+    async def _get_frame_and_put_to_redis(
+        self, redis_uuid: str, response: ClientResponse
+    ):
         jpeg_bytes = await get_next_jpeg(response)
-        self.uuid_setter(image_uuid := str(uuid.uuid4()))
-        img = Image.open(io.BytesIO(jpeg_bytes))
-        image_data = pickle.dumps(np.asarray(img))
```

Comment:
Before we move forward and forget would like 5 minutes look and maybe A comment left in the code why we make this decision now (allo5wing to change in the future if necessary). 

First we decided to not store numpy arrays of the raw images (jpg decompressed) after realising it would occupy enormous amounts of DB space with no real benefit on speed. 

Question on current implementation:  there 3 ways to store binary jpg into redis

I) directly as an bytesIO object using data.getvalues()
II) serialising it (as we make here with pickle)
III) using a redis optimize structure called redis bitmap for such objects

Happy we stay with (II) as I don't fully know disadvantages of the other ones Vs this one but maybe leave a comment this could be relooked if necessary in the future?

Reply:
Issue at https://github.com/DiamondLightSource/mx-bluesky/issues/592 to investigate

Reply:
Ah, I actually think this PR already removes the pickling and just stores the raw bytes. I will add a comment though.